src/middlewared/middlewared/plugins/mail.py

```diff
@@ -110,13 +110,31 @@ class Config:
         verrors = ValidationErrors()
 
         if new['smtp'] and new['user'] == '':
-            verrors.add('mail_update.user', 'This field is required when SMTP authentication is enabled')
+            verrors.add(
+                'mail_update.user',
+                'This field is required when SMTP authentication is enabled',
+            )
+
+        self.__password_verify(new['pass'], 'mail_update.pass', verrors)
 
         if verrors:
             raise verrors
 
         await self.middleware.call('datastore.update', 'system.email', config['id'], new, {'prefix': 'em_'})
-        return config
+        return await self.config()
+
+    def __password_verify(self, password, schema, verrors=None):
+        if not password:
+            return
+        if verrors is None:
+            verrors = ValidationErrors()
+        # FIXME: smtplib does not support non-ascii password yet
+        # https://github.com/python/cpython/pull/8938
+        try:
+            password.encode('ascii')
+        except UnicodeEncodeError:
+            verrors.add(schema, 'At this time only ASCII passwords are allowed.')
```

Comment:
@wblock could you please verify if this error is to your liking?

Reply:
Suggested:
`` verrors.add(schema, 'UTF characters are not allowed in passwords.') ``

Reply:
@wblock UTF is not the most correct term to use here IMO. I fear some people dont even know what that means. Can we stick to ASCII or say something along the lines of non special composed characters or so?

Reply:
I agree, and had similar concerns about about "ASCII".  Until I looked up the context, I was thinking it might be about plain text versus encrypted passwords.

We can be more explicit: "Only plain text characters (8-bit ASCII) are allowed in passwords. UTF or composed characters are not allowed."

Reply:
Sounds good. Thanks